contracts/interfaces/IEverlong.sol

```diff
@@ -1,31 +1,46 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.20;
 
-import { IERC4626 } from "openzeppelin/interfaces/IERC4626.sol";
+import { ERC4626 } from "solady/tokens/ERC4626.sol";
 import { IEverlongAdmin } from "./IEverlongAdmin.sol";
 import { IEverlongEvents } from "./IEverlongEvents.sol";
-import { IEverlongPositions } from "./IEverlongPositions.sol";
+import { IEverlongPortfolio } from "./IEverlongPortfolio.sol";
 
-interface IEverlong is
+abstract contract IEverlong is
+    ERC4626,
     IEverlongAdmin,
-    IERC4626,
     IEverlongEvents,
-    IEverlongPositions
+    IEverlongPortfolio
 {
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Structs                                                 │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @notice Tracks the total amount of bonds managed by Everlong
+    ///         with the same maturity.
+    struct Position {
+        /// @notice Time when the position matures.
+        uint128 maturityTime;
+        /// @notice Quantity of bonds in the position.
+        uint128 bondAmount;
+        /// @notice Vault share price at time of purchase.
+        uint128 vaultSharePrice;
```

Comment:
If this is used for negative interest, we should just use the vault share price from Hyperdrive’s checkpoint. Otherwise, this may not be needed at all. Either way, we don’t need to store it here.

Reply:
Yeah so with negative interest and using the checkpoint vault share price I see a pretty massive difference in estimated vs actual output...

```
 Want: 10239305534593637769923
 Have: 9986496180175740679947
```

However when I use the actual purchase vault share price, I still see a difference, but it's massively less:

```
Want: 9987830847237632187943
Have: 9987830847237632186046
```

Unsure of why this is, maybe I'm pulling the price from the wrong checkpoint?

Reply:
Now have down to the wei accurate estimates